src/controllers/middlewares/jwtVerification.js

```diff
@@ -0,0 +1,16 @@
+const jwt = require('jsonwebtoken');
+
+exports.verify = (req, res, next) => {
+  const token = req.cookies.jwt;
+  if (token) {
+    jwt.verify(token, process.env.SECRET, (err, decoded) => {
+      if (decoded) {
+        req.auth = decoded;
+        next();
+      }
+    });
+  } else {
+    req.auth = false;
+    next();
```

Comment:
Another approach you could take is to have this `verify` middleware *only* used on the routes that require auth (instead of every route), and then instead of setting `req.auth` to true or false, it would just stop the request with a 401 unauthorized.
Then you would have to pick and choose which routes use this middleware, instead of just getting to say `router.use` once, but the advantage is that then every route doesn't need to individually have an if-statement to look at req.auth and fail if it's false.